### PR TITLE
Project timeline upgrades

### PR DESCRIPTION
**Pull request by peripheralist**
Overhaul of the project timeline chart:
- Improves performance by replacing a series of subgraph queries with a single aliased subgraph query
- Timeline resolution increased from 20 to 30 data points
- Adds new "Trending" view to timeline, displaying trending score over time
- Big housekeeping effort across the UI components
- Style improvements

_Note:_ The chart still occasionally fails to load the "1 year" window, due to a subgraph error. My best guess is this is related to querying data for much older block heights, which indexers may not have as readily available.

### CODE REVIEW

File: src/components/VolumeChart/components/TimelineChart.tsx [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+  const yDomain: [number, number] =
+    view === 'trendingScore' && highTrendingScore
+      ? [
+          defaultYDomain[0],
+          Math.max(highTrendingScore, highTrendingPoint) * 1.05,
+        ]
+      : defaultYDomain
+
+  const {
+    theme: { colors },
```
Comment (peripheralist) [inline src/components/VolumeChart/components/TimelineChart.tsx:63]: i guess this is cool?

File: src/components/VolumeChart/hooks/useProjectTimeline.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+          { block: number; timestamp: number },
+        ]
+      >
+    },
+    {
+      staleTime: minutesToMS(5),
+    },
+  )
+
+  const { blocks, timestamps } = useMemo(() => {
```
Comment (peripheralist) [inline src/components/VolumeChart/hooks/useProjectTimeline.ts:47]: here we compute the blockNumbers to pass to the new graphql query

File: src/components/VolumeChart/hooks/useProjectTimeline.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    for (let i = 0; i < COUNT; i++) {
+      const point = queryResult[`p${i}` as keyof typeof queryResult]
+
+      if (!point) continue
+
+      points.push({
+        timestamp: timestamps[i],
+        trendingScore: floatFromWad(point.trendingScore),
+        balance: floatFromWad(point.currentBalance),
+        volume: floatFromWad(point.volume),
```
Comment (peripheralist) [inline src/components/VolumeChart/hooks/useProjectTimeline.ts:94]: converting these values to `number` floats is the only way we can make them compatible with the recharts components

File: src/components/inputs/JuiceListbox.tsx [hunk 4]
```diff
@@ -28,7 +28,7 @@ export function JuiceListbox<T>(props: JuiceListboxProps<T>) {
           )}
         >
           <span className="block truncate">{value?.label}</span>
-          <span className="pointer-events-none flex items-center justify-end pr-2">
```
Comment (peripheralist) [inline src/components/inputs/JuiceListbox.tsx:31]: random style fix, hoping this wasn't intentional? @wraeth-eth

File: src/graphql/projects/projectTL.graphql [hunk 5]
```diff
@@ -0,0 +1,130 @@
+fragment ProjectTLParts on Project {
+  currentBalance
+  volume
+  trendingScore
+}
+
+query ProjectTL(
+  $id: ID!
+  $block0: Int
```
Comment (peripheralist) [inline src/graphql/projects/projectTL.graphql:9]: yes this is gross, but it works well. using aliases lets us effectively make multiple unique queries in a single http req

here we want to get project data at 30 different historical block heights. unfortunately graphql doesn't allow for any logic or iterators, so we have to provide each block height as a named argument 🥲

File: src/components/VolumeChart/components/TimelineChart.tsx (hunk 1 above)
Comment (tomquirk) [reply src/components/VolumeChart/components/TimelineChart.tsx:63]: nicceee find. lfg

File: src/components/VolumeChart/components/TimelineChart.tsx [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+        number,
+        number,
+      ],
+    [range],
+  )
+
+  const stroke = themeOption === 'dark' ? colors.slate[300] : colors.grey[400]
+  const color = themeOption === 'dark' ? colors.slate[200] : colors.grey[400]
+  const bg = themeOption === 'dark' ? colors.slate[900] : 'white'
+  const fontSize = '0.7rem'
```
Comment (tomquirk) [inline src/components/VolumeChart/components/TimelineChart.tsx:78]: nit: does 0.75rem look ok? to match tailwind `text-xs`

File: src/graphql/projects/projectTL.graphql [hunk 7]
```diff
@@ -0,0 +1,130 @@
+fragment ProjectTLParts on Project {
+  currentBalance
+  volume
+  trendingScore
+}
+
+query ProjectTL(
```
Comment (tomquirk) [inline src/graphql/projects/projectTL.graphql:7]: **suggestion** maybe `ProjectTimeline` for maximum clarity

File: src/graphql/projects/projectTL.graphql (hunk 5 above)
Comment (tomquirk) [reply src/graphql/projects/projectTL.graphql:9]: lgtm

File: src/utils/format/formatNumber.ts [hunk 8]
```diff
@@ -74,6 +74,18 @@ export const formatWad = (
...
 
+/**
+ * Returns a float representation of a given [wadValue]
+ *
+ * @example
+ * // returns 1.23
+ * fromWad(1230000000000000000);
+ *
+ */
+export const floatFromWad = (wadValue?: BigNumberish) => {
```
Comment (tomquirk) [inline src/utils/format/formatNumber.ts:85]:
```suggestion
export const wadToFloat = (wadValue?: BigNumberish) => {
```

**Review by tomquirk — APPROVED** [review]
@peripheralist pre-approving with a few suggestions 🙏

**Comment by peripheralist** [comment]
thanks @aeolianeth , changes implemented 🫡

**Pull request merged**